### PR TITLE
[fix](DOE)Fix es p0 case error

### PR DESCRIPTION
**Pull request by qidaye**
# Proposed changes

Fix es array parse error, introduced by #16806

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by qidaye** [comment]
es_p0 case can not passed before this bug #17491 is solved.

**Comment by qidaye** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java [hunk 1]
```diff
@@ -169,13 +181,15 @@ public static List<Column> genColumnsFromEs(EsRestClient client, String indexNam
             boolean mappingEsId) {
         String mapping = client.getMapping(indexName);
         ObjectNode mappings = getMapping(mapping);
-        ObjectNode rootSchema = getRootSchema(mappings, mappingType);
-        return genColumnsFromEs(indexName, mappingType, rootSchema, mappingEsId);
+        // Get array_fields while removing _meta property.
+        List<String> arrayFields = new ArrayList<>();
+        ObjectNode rootSchema = getRootSchema(mappings, mappingType, arrayFields);
```
Comment (SaintBacchus) [inline fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java:186]: This methon is a bit like C++ style. How about this?
```suggestion
        List<String> arrayFields = extractArrayFields(mappings, mappingType);
        ObjectNode rootSchema = getRootSchema(mappings, mappingType);
```

Comment (qidaye) [reply fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java:186]: We can only get `array_fields` during the `getRootSchema` before the `_meta` field is removed.
The suggestion changes are more like the style before the refactor, which will add ES version judgement in the `extractArrayFields` and will be duplicated with the logical in `getRootSchema`.
You can check the old style via git history.

**Comment by SaintBacchus** [comment]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**